[기능]카카오 로그인 토큰 발급 구현

choi-day opened the pull request:
## 🔖 PR 타입

해당 PR의 성격에 맞게 체크해주세요.

- [x] feat (새로운 기능 추가)
- [ ] fix (버그 수정)
- [ ] chore (빌드/설정/배포 등 비즈니스 로직 외 작업)
- [ ] refactor (리팩토링, 로직 변화 없음)
- [ ] docs (문서 작업)
- [ ] style (포맷팅, 세미콜론 누락 등)
- [ ] test (테스트 추가 또는 수정)

---

## 📄 주요 변경 사항
변경한 핵심 내용을 bullet 형식으로 정리해주세요.

- 패키지명을 변경하였습니다.
- 인터넷 사용 권한을 부여하였습니다.
- 앱 전역 설정을 위해 application클래스를 상속받은 MyApplication.kt를 app폴더 아래 생성하였습니다.
- 카카오 로그인을 클릭하면 사용자의 동의를 받는 화면으로 넘어간 후 동의를 받아 로그인이 완료되면 토큰을 얻습니다.

---

## 🧪 테스트 내용
직접 확인한 동작 또는 QA 기준 테스트 목록을 작성해주세요.

- [x] 버튼 클릭 시 카카오 로그인 연결
- [x] 카카오톡이 설치 되어있을 때 로그인 후 토큰 발급 확인
- [x] 카카오톡이 설치되어있지 않을 때 로그인 후 토큰 발급 확인

---

## 📱 스크린샷 / 영상 (선택)

[Screen_recording_20250723_235701.webm](https://github.com/user-attachments/assets/2aca57df-2d6e-4bcb-bf30-54b382af8dde)


---

## 🧩 관련 이슈 / 작업 항목

- Resolved: #3 
- 관련 작업: 카카오 로그인 토큰 발급 확인

---

## ✅ 체크리스트

- [ ] CI/CD 통과 확인
- [x] 커밋 메시지 컨벤션 (`feat:`, `fix:` 등) 적용
- [x] main 브랜치로 바로 PR 올리지 않음 (ex. dev → main은 릴리즈 시에만)
- [x] 불필요한 로그 / 디버깅 코드 제거

---

## 👀 리뷰어 참고사항 (중요)

- [ ] 앱 전역 설정을 위해 application클래스를 상속받은 MyApplication.kt를 app폴더 아래 생성하였습니다. 앞으로 앱 전역 설정이 필요하다면 해당 파일에 구현하는 것이 좋을 것 같아요.
- [ ] 카카오 로그인 구현을 위해 그래들 설정을 많이 추가하였습니다 병합시 주의해야할 것 같아요.
- [ ] local.properties 내부 내용은 올리지 않았어요. 어디에 보낼 지 애매해서 아직 내용을 안 보냈습니다. 필요할 때 말씀해주세요.

---

## 🧼 임시 코드 또는 추후 개선 예정

- 서버와 연결
- 로그인 후 기존 사용자인지 판단
- 기존 사용자면 홈화면으로 새로운 사용자면 회원가입 화면으로 이동

choi-day committed:
#3 chore: 인터넷 사용 권한 부여

choi-day committed:
#3 chore: 카카오 로그인을 위한 maven 레파지토리 설정

choi-day committed:
#3 chore: 패키지명 변경

choi-day committed:
#3 chore: 로그인을 위한 그래들 설정

choi-day committed:
#3 chore: 인가코드를 받기 위한 액티비티 추가

choi-day committed:
#3 chore: 오타수정

choi-day committed:
#3 chore: 테마 설정 수정

choi-day committed:
#3 chore: buildconfig사용을 위한 의존성 추가

choi-day committed:
#3 feat: MyApplication.kt 추가 및 kakaoSdk 초기화 구현

choi-day committed:
#3 feat: 카카오 로그인 구현

choi-day committed:
#3 feat: 카카오 로그인 테스트를 위한 토큰 확인 구현

choi-day committed:
#3 chore: AndroidManifest 에 MyApplication.kt등록

choi-day committed:
merge: Merge remote-tracking branch 'origin/dev' into feature/login

choi-day committed:
merge: Merge branch 'feature/login' of https://github.com/choi-day/ddu-ru-mobile into feature/login

hyeminililo commented on app/src/main/java/com/gildongmu/ddu_ru_mobile/MyApplication.kt:
```diff
@@ -0,0 +1,12 @@
+class MyApplication : Application() {
+    override fun onCreate() {
+        super.onCreate()
+        KakaoSdk.init(this, BuildConfig.KAKAO_NATIVE_APP_KEY)
```
sdk 초기화 관련하여 자동로그인 참고해서 수정부탁드립니다 :)

choi-day committed:
#3 chore : 버전 오류에 따른 코드 수정

choi-day committed:
#3 feat : margin수정

choi-day committed:
#3 chore: 서버 통신을 위한 권한 설정 및 테스트 코드 작성

choi-day committed:
#3 chore: 서버 통신을 위한 의존성 추가

choi-day committed:
#3 feat: 통신을 위한 dto추가

choi-day committed:
#3 feat: 토큰 발급 후 서버에 Post구현

choi-day committed:
#6 fix : baseUrl 수정

choi-day committed:
#3 chore: gitignore 설정

choi-day committed:
#3 Merge: merge branch 'feature/login' of https://github.com/GIL-DONG-MU/ddu-ru-mobile into feature/login

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/com/gildongmu/ddu_ru_mobile/ui/LoginActivity.kt
#	app/src/main/java/com/gildongmu/ddu_ru_mobile/util/login/GoogleLoginHelper.kt
#	app/src/main/java/com/gildongmu/ddu_ru_mobile/util/login/KakaoLoginHelper.kt
#	app/src/main/res/values/themes.xml

choi-day committed:
#3 fix : buildConfig관련 오류 수정

choi-day committed:
#3 fix : buildConfig관련 오류 수정
